src/Lean/Parser/Tactic/Doc.lean

```diff
@@ -0,0 +1,217 @@
+Declare this tactic to be an alias or alternative form of an existing tactic.
+
+This has the following effects:
+ * The alias relationship is saved
+ * The docstring is taken from the original tactic, if present
+-/
+syntax (name := tactic_alias) "tactic_alias" (ppSpace ident) : attr
+
+builtin_initialize
+  let name := `tactic_alias
+  registerBuiltinAttribute {
+    name := name,
+    ref := by exact decl_name%,
+    add := fun decl stx kind => do
+      unless kind == AttributeKind.global do throwError "invalid attribute '{name}', must be global"
+      unless ((← getEnv).getModuleIdxFor? decl).isNone do
+        throwError "invalid attribute '{name}', declaration is in an imported module"
+      let `(attr|tactic_alias $tgt:ident) := stx
+        | throwError "invalid '{name}' attribute"
+
+      let tgtName ← Lean.Elab.realizeGlobalConstNoOverloadWithInfo tgt
+
+      if let some tgt' ← aliasOfTactic tgtName then
+        throwError "'{tgtName}' is itself an alias of '{tgt'}'"
+      modifyEnv fun env => tacticAliasExt.addEntry env (decl, tgtName)
+      if let some docs ← findDocString? (← getEnv) tgtName then
+        if (← findDocString? (← getEnv) decl).isSome then
+          logWarningAt stx m!"Replacing docstring for '{decl}' with the one from '{tgtName}'"
+        addDocString decl docs
+    descr := "Register a tactic parser as an alias of an existing tactic, so they can be grouped together in documentation.",
+    applicationTime := .afterTypeChecking
+  }
+
+/--
+The known tactic tags that allow tactics to be grouped by purpose.
+-/
+builtin_initialize knownTacticTagExt
+    : PersistentEnvExtension
+        (Name × String × Option String)
+        (Name × String × Option String)
+        (NameMap (String × Option String)) ←
+  registerPersistentEnvExtension {
+    mkInitial := pure {},
+    addImportedFn := fun _ => pure {},
+    addEntryFn := fun as (src, tgt) => as.insert src tgt,
+    exportEntriesFn := fun es =>
+      es.fold (fun a src tgt => a.push (src, tgt)) #[] |>.qsort (Name.quickLt ·.1 ·.1)
+  }
+
+/--
+Get the user-facing name and docstring for a tactic tag.
+-/
+def tagInfo [Monad m] [MonadEnv m] (tag : Name) : m (Option (String × Option String)) := do
+  let env ← getEnv
+  match env.getModuleIdxFor? tag with
+  | some modIdx =>
+    match (knownTacticTagExt.getModuleEntries env modIdx).binSearch (tag, default) (Name.quickLt ·.1 ·.1) with
+    | some (_, val) => pure (some val)
+    | none => pure none
+  | none => pure (knownTacticTagExt.getState env |>.find? tag)
+
+/-- Enumerate the tactic tags that are available -/
+def allTags [Monad m] [MonadEnv m] : m (List Name) := do
+  let env ← getEnv
+  let mut found : NameSet := {}
+  for (tag, _) in knownTacticTagExt.getState env do
+    found := found.insert tag
+  for arr in tacticAliasExt.toEnvExtension.getState env |>.importedEntries do
+    for (tag, _) in arr do
+      found := found.insert tag
+  pure (found.toArray.qsort (·.toString < ·.toString) |>.toList)
+
```

Comment:
```suggestion

elab "#print_tactic_tags" : command => do
  logInfo <| "\n".intercalate <| (← allTags).map toString

```

This, or some variant, will be a necessity for anyone adding docs -- gotta see what tags are available!

Reply:
I added a fancier version that also shows user-facing names, docstrings for tags, and the tactics that are included in them (under `#print tactic tags`). Thanks!